### PR TITLE
Migrate departments to new DB

### PR DESCRIPTION
**Pull request by KaiSnorlax**
Removed unnecessary endpoint, and changed program interface to reflect database.

### CODE REVIEW

File: src/api/departments.ts [hunk 1]
```diff
@@ -1,53 +1,39 @@
...
   id: number;
+  department_id: number;
   name: string;
   type: string;
 }
 
 router.get("/departments", async (req, res) => {
-  const departments = await sql<Department[]>`SELECT id, name, college_prefix FROM departments`;
+  const departments = await sql<Department[]>`
+    SELECT id, name FROM departments`;
```
Comment (Nathan-Lackie) [inline src/api/departments.ts]: What's up with the spacing here?

Comment (Nathan-Lackie) [reply src/api/departments.ts]: I'd prefer to put the closing ` on it's own line

File: src/api/departments.ts [hunk 2]
```diff
@@ -1,53 +1,39 @@
...
-    name: department.name,
-    college: department.college_prefix,
-  });
-});
-
 router.get("/departments/:departmentId/programs", async (req, res) => {
   const departmentId = req.params.departmentId;
 
-  const programs = await sql<Program[]>`
+  const programs = await sql<Omit<Program[], "department_id">>`
```
Comment (Nathan-Lackie) [inline src/api/departments.ts]: Why are we adding a new field just to omit it here?

Comment (KaiSnorlax) [reply src/api/departments.ts]: I believe our interfaces should reflect our database, and we can use Pick and Omit when needed.

Comment (Nathan-Lackie) [reply src/api/departments.ts]: Not a bad idea actually, before I was using interfaces for the result of each individual query.
If we do that though, we should pull the database type definitions out into their own file/files.

Comment (KaiSnorlax) [reply src/api/departments.ts]: I like that idea, could you open an issue for that?

File: src/api/departments.ts [hunk 3]
```diff
@@ -1,50 +1,38 @@
...
```
Comment (Nathan-Lackie) [inline src/api/departments.ts]: Actually, I think we should always use Pick<> here, as it closer reflects the SQL query.

Comment (Nathan-Lackie) [reply src/api/departments.ts]: Believe you can use a union type `Pick<Program, "id" | "name" | "type">[]`

**Review by Nathan-Lackie — APPROVED** [review]

**Pull request merged**